diff --git a/modules/objdetect/src/detection_based_tracker.cpp b/modules/objdetect/src/detection_based_tracker.cpp
index 0cdcaface..8260f5c26 100644
--- a/modules/objdetect/src/detection_based_tracker.cpp
+++ b/modules/objdetect/src/detection_based_tracker.cpp
@@ -403,6 +403,7 @@ void cv::DetectionBasedTracker::SeparateDetectionWork::workcycleObjectDetector()
 
         int64 t1_detect=getTickCount();
 
+        LOGD("Calling detect");
         cascadeInThread->detect(imageSeparateDetecting, objects);
 
         /*cascadeInThread.detectMultiScale( imageSeparateDetecting, objects,
@@ -588,7 +589,7 @@ cv::DetectionBasedTracker::Parameters::Parameters()
 cv::DetectionBasedTracker::InnerParameters::InnerParameters()
 {
     numLastPositionsToTrack=4;
-    numStepsToWaitBeforeFirstShow=6;
+    numStepsToWaitBeforeFirstShow=0;
     numStepsToTrackWithoutDetectingIfObjectHasNotBeenShown=3;
     numStepsToShowWithoutDetecting=3;
 
@@ -626,6 +627,7 @@ cv::DetectionBasedTracker::~DetectionBasedTracker()
 
 void DetectionBasedTracker::process(const Mat& imageGray)
 {
+    LOGD("DetectionBasedTracker::process() -- start");
     CV_INSTRUMENT_REGION()
 
     CV_Assert(imageGray.type()==CV_8UC1);
